src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+                let minimum_size = minimum_size.assume_init();
+                let natural_size = natural_size.assume_init();
+                (minimum_size, natural_size)
+            } else {
+                (-1, -1)
+            }
+        }
+    }
+
+    fn parent_get_aligned_area<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        flags: CellRendererState,
+        cell_area: &gdk::Rectangle,
+    ) -> gdk::Rectangle {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            let mut aligned_area = gdk::Rectangle::uninitialized();
+            if let Some(f) = (*parent_class).get_aligned_area {
+                f(
+                    renderer.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    flags.to_glib(),
+                    cell_area.to_glib_none().0,
+                    aligned_area.to_glib_none_mut().0,
+                );
+            }
+            aligned_area
+        }
+    }
+
+    fn parent_render<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        cr: &cairo::Context,
+        widget: &P,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).render {
+                f(
+                    renderer.to_glib_none().0,
+                    cr.to_glib_none().0,
+                    widget.as_ref().to_glib_none().0,
+                    background_area.to_glib_none().0,
+                    cell_area.to_glib_none().0,
+                    flags.to_glib(),
+                )
+            }
+        }
+    }
+
+    fn parent_activate<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: &gdk::Event,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> bool {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).activate {
+                from_glib(f(
+                    renderer.to_glib_none().0,
+                    mut_override(event.to_glib_none().0),
+                    widget.as_ref().to_glib_none().0,
+                    path.to_glib_none().0,
+                    background_area.to_glib_none().0,
+                    cell_area.to_glib_none().0,
+                    flags.to_glib(),
+                ))
+            } else {
+                false
```

Comment:
`gtk_cell_renderer_activate()` returns `FALSE` if it's `NULL`, so this is indeed the right behaviour :)